pkg/ps/ps.go

```diff
@@ -37,6 +38,14 @@ func GetContainerLists(runtime *libpod.Runtime, options entities.ContainerListOp
 				return nil, err
 			}
 			filterFuncs = append(filterFuncs, generatedFunc)
+
+			if options.External {
+				generatedExtFunc, err := filters.GenerateExternalContainerFilterFuncs(k, v, runtime)
+				if err != nil {
+					return nil, err
+				}
+				filterExtFuncs = append(filterExtFuncs, generatedExtFunc)
+			}
```

Comment:
This will cause regressions, GenerateExternalContainerFilterFuncs() will error on unknown options which means `podman ps --external --filter ...`   will break all other existing filters which does not seem right.

I think design wise it would go long way if we could abstract the same filters for both normal and external container so we do not need the different code handling filters at all.

Reply:
> This will cause regressions, GenerateExternalContainerFilterFuncs() will error on unknown options which means `podman ps --external --filter ...` will break all other existing filters which does not seem right.
> 
> I think design wise it would go long way if we could abstract the same filters for both normal and external container so we do not need the different code handling filters at all.

I was thinking to implement `GenerateFilerFunc` as based on generics, however some types which are used in the logic can't be implemented as generics ones. 

Do you want me to implement all remaining filters features for external containers?